front/src/common/Map/Buttons/MapButtons.js

```diff
@@ -12,28 +12,68 @@ import ButtonFullscreen from 'common/ButtonFullscreen';
 import MapSearch from 'common/Map/Search/MapSearch';
 import MapSettings from 'common/Map/Settings/MapSettings';
 import MapKey from 'common/Map/MapKey';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateFeatureInfoClickOSRD } from 'reducers/osrdconf';
+
+import { getFeatureInfoClick } from 'reducers/osrdconf/selectors';
+import useOutsideClick from 'utils/hooks/useOutsideClick';
 import ButtonMapInfras from './ButtonMapInfras';
 
 export default function MapButtons(props) {
   const { resetPitchBearing, withInfraButton } = props;
-  const [showSearch, setShowSearch] = useState(false);
-  const [showSettings, setShowSettings] = useState(false);
-  const [showMapKey, setShowMapKey] = useState(false);
+  const featureInfoClick = useSelector(getFeatureInfoClick);
+  const dispatch = useDispatch();
+
+  const MAP_POPOVERS = { SEARCH: 'SEARCH', SETTINGS: 'SETTINGS', KEY: 'KEY' };
```

Comment:
Can you instantiate this dict outside the component ? This way, it won't be re-created each time the component re-render

Reply:
Done